### PR TITLE
Adding ovirt-engine-collect-logs role

### LINKED ISSUE #51: Add role to collect logs from systems
There should be task, which is able to collect relevant logs from systems which are used as part of ovirt deployment.

* engine
* hypervisors
* db
* dwh

### PR DESCRIPTION
**Pull request by lukas-bednar**
Fixes #51

### CODE REVIEW

**Comment by lukas-bednar** [comment]
@StLuke @pkubica FYI

**Comment by lukas-bednar** [comment]
Verified for engine type:

```
drwxr-xr-x root/root         0 2016-08-17 11:36 var/tmp/ovirt-engine-logs/
-rw-r--r-- root/root      9749 2016-08-17 11:36 var/tmp/ovirt-engine-logs/pstree.txt
-rw-r--r-- root/root      5295 2016-08-17 11:36 var/tmp/ovirt-engine-logs/iptables.txt
-rw------- root/root    438701 2016-08-17 11:35 var/tmp/ovirt-engine-logs/messages
-rw-r--r-- root/root  17464691 2016-08-17 11:36 var/tmp/ovirt-engine-logs/lsof.txt
drwxr-xr-x ovirt/ovirt       0 2016-08-16 09:38 var/tmp/ovirt-engine-logs/ovirt-engine/
drwxr-xr-x ovirt/ovirt       0 2016-07-01 13:12 var/tmp/ovirt-engine-logs/ovirt-engine/notifier/
drwxr-xr-x ovirt/ovirt       0 2016-07-01 13:12 var/tmp/ovirt-engine-logs/ovirt-engine/host-deploy/
drwxr-xr-x root/root         0 2016-06-01 11:09 var/tmp/ovirt-engine-logs/ovirt-engine/ovirt-image-uploader/
-rw-r--r-- ovirt/ovirt    1153 2016-08-16 09:38 var/tmp/ovirt-engine-logs/ovirt-engine/boot.log
-rw-r--r-- ovirt/ovirt       0 2016-08-16 09:38 var/tmp/ovirt-engine-logs/ovirt-engine/console.log
-rw-r--r-- ovirt/ovirt   36870 2016-08-16 10:15 var/tmp/ovirt-engine-logs/ovirt-engine/server.log
-rw-r--r-- ovirt/ovirt       0 2016-08-16 09:38 var/tmp/ovirt-engine-logs/ovirt-engine/ui.log
drwxr-xr-x root/root         0 2016-06-01 10:18 var/tmp/ovirt-engine-logs/ovirt-engine/ovirt-log-collector/
-rw-r--r-- ovirt/ovirt   83414 2016-08-17 10:40 var/tmp/ovirt-engine-logs/ovirt-engine/engine.log
drwxr-xr-x ovirt/ovirt       0 2016-07-01 13:12 var/tmp/ovirt-engine-logs/ovirt-engine/dump/
drwxr-xr-x root/root         0 2016-08-16 09:28 var/tmp/ovirt-engine-logs/ovirt-engine/setup/
-rw-r--r-- root/root   2830873 2016-08-16 09:38 var/tmp/ovirt-engine-logs/ovirt-engine/setup/ovirt-engine-setup-20160816102812-0c4xqu.log
drwxr-xr-x root/root         0 2016-06-01 10:45 var/tmp/ovirt-engine-logs/ovirt-engine/ovirt-iso-uploader/
-rw-r--r-- root/root     46926 2016-08-17 11:36 var/tmp/ovirt-engine-logs/rpm-list.txt
-rw-r--r-- root/root    101381 2016-08-17 11:36 var/tmp/ovirt-engine-logs/yum-list.txt
-rw-r--r-- root/root       634 2016-08-17 11:36 var/tmp/ovirt-engine-logs/lspci.txt
-rw-r--r-- root/root      8054 2016-08-17 11:36 var/tmp/ovirt-engine-logs/netstat.txt
-rw-r--r-- root/root     28249 2016-08-17 11:36 var/tmp/ovirt-engine-logs/sysctl.txt
-rw-r--r-- root/root       900 2016-08-17 11:36 var/tmp/ovirt-engine-logs/services.txt
-rw-r--r-- root/root      3164 2016-08-17 11:36 var/tmp/ovirt-engine-logs/lsmod.txt
```

File: roles/ovirt-engine-collect-logs/tasks/hypervisor.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+---
+- name: Link vdsm and libfirt logs
+  file:
+    src: "{{ item.src }}"
+    dest: "{{ ovirt_engine_collect_logs_tmp_dir }}/{{ item.dest }}"
+    state: link
+  with_items:
+    - { src: "/var/log/vdsm", dest: "vdsm" }
+    - { src: "/var/log/libvirt", dest: "libvirt" }
```
Comment (StLuke) [inline roles/ovirt-engine-collect-logs/tasks/hypervisor.yml]: add libvirt dump of vm cores (possibly optional as large VMs might be present)

Comment (StLuke) [inline roles/ovirt-engine-collect-logs/tasks/hypervisor.yml]: add engine ovirt_engine_history databas dumps (possibly optional as large dbs might be present)

Comment (lukas-bednar) [inline roles/ovirt-engine-collect-logs/tasks/hypervisor.yml]: core dumps already included as part of libvirt directory

ovirt_engine_history will be added as part of DWH system.

**Comment by lukas-bednar** [comment]
verified!

**Comment by pkubica** [comment]
missing dwh logs ?

File: roles/ovirt-collect-logs/tasks/dwh.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+---
+- name: Link DWH logs
+  file:
+    src: "{{ item.src }}"
+    dest: "{{ ovirt_collect_logs_tmp_dir }}/{{ item.dest }}"
+    state: link
+  with_items:
+    - { src: "/var/log/ovirt-engine-dwh", dest: "ovirt-engine-dwh-logs" }
+    - { src: "/var/lib/ovirt-engine-dwh", dest: "ovirt-engine-dwh-data" }
```
Comment (StLuke) [inline roles/ovirt-collect-logs/tasks/dwh.yml]: add /etc/ovirt-engine-dwh/*

Comment (lukas-bednar) [inline roles/ovirt-collect-logs/tasks/dwh.yml]: done

**Comment by lukas-bednar** [comment]
> missing dwh logs ?
just not installed

File: roles/ovirt-collect-logs/tasks/engine.yml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+---
+- name: Link ovirt-engine logs
+  file:
+    src: "{{ item.src }}"
+    dest: "{{ ovirt_collect_logs_tmp_dir }}/{{ item.dest }}"
+    state: link
+  with_items:
+    - { src: "/var/log/ovirt-engine", dest: "ovirt-engine-logs" }
+    - { src: "/var/lib/ovirt-engine", dest: "ovirt-engine-data" }
+    - { src: "/etc/ovirt-engine", dest: "ovirt-engine-etc" }
```
Comment (StLuke) [inline roles/ovirt-collect-logs/tasks/engine.yml]: /etc/ovirt-engine-setup.conf.d
/etc/ovirt-host-deploy.conf.d

File: roles/ovirt-collect-logs/tasks/hypervisor.yml [hunk 4]
```diff
@@ -0,0 +1,21 @@
+---
+- name: Link vdsm and libfirt logs
+  file:
+    src: "{{ item.src }}"
+    dest: "{{ ovirt_collect_logs_tmp_dir }}/{{ item.dest }}"
+    state: link
+  with_items:
+    - { src: "/var/log/vdsm", dest: "vdsm" }
+    - { src: "/var/log/libvirt", dest: "libvirt" }
```
Comment (StLuke) [inline roles/ovirt-collect-logs/tasks/hypervisor.yml]: /etc/ovirt-vmconsole
/etc/vdsm

File: roles/ovirt-collect-logs/tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+  shell: "netstat -lnp &> {{ ovirt_collect_logs_tmp_dir }}/netstat.txt"
+  ignore_errors: true
+
+- name: Dump lsmod
+  shell: "lsmod &> {{ ovirt_collect_logs_tmp_dir }}/lsmod.txt"
+  ignore_errors: true
+
+- name: Dump lspci
+  shell: "lspci &> {{ ovirt_collect_logs_tmp_dir }}/lspci.txt"
+  ignore_errors: true
```
Comment (StLuke) [inline roles/ovirt-collect-logs/tasks/main.yml]: All dumps I would add to new role, to be able to run it on hypervisors, remote dwh/database, servers, engine

Comment (lukas-bednar) [inline roles/ovirt-collect-logs/tasks/main.yml]: I don't understand comment, it is able to run on them. please be more specific

File: roles/ovirt-collect-logs/tasks/hypervisor.yml (hunk 4 above)
Comment (lukas-bednar) [inline roles/ovirt-collect-logs/tasks/hypervisor.yml]: done

File: roles/ovirt-collect-logs/tasks/engine.yml (hunk 3 above)
Comment (lukas-bednar) [inline roles/ovirt-collect-logs/tasks/engine.yml]: done

**Comment by lukas-bednar** [comment]
verified

File: roles/ovirt-collect-logs/tasks/main.yml (hunk 5 above)
Comment (StLuke) [inline roles/ovirt-collect-logs/tasks/main.yml]: I see now, nevermind

**Comment by StLuke** [comment]
+1

**Pull request merged**